lib/full_url_for.js

```diff
@@ -4,7 +4,8 @@ const { parse, URL } = require('url');
 const encodeURL = require('./encode_url');
 
 function fullUrlForHelper(path = '/') {
-  if (path.startsWith('//')) return path;
+  const pathRegex = /^(\/\/|http(s)?:)/;
```

Comment:
not sure if `ignore case` flag is needed, I don't mind adding it. I benchmarked the flag and I didn't notice any difference.